core/src/main/java/org/apache/accumulo/core/client/summary/summarizers/EntryLengthSummarizer.java

```diff
@@ -0,0 +1,159 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.client.summary.summarizers;
+
+import java.math.RoundingMode;
+import java.util.Map;
+
+import org.apache.accumulo.core.client.summary.Summarizer;
+import org.apache.accumulo.core.client.summary.SummarizerConfiguration;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+
+import com.google.common.math.IntMath;
+
+/**
+ * Summarizer that computes summary information about field lengths.
+ * Specifically key length, row length, family length, qualifier length, visibility length, and value length.
+ * Incrementally computes minimum, maximum, count, sum, and log2 histogram of the lengths.
+ */
+public class EntryLengthSummarizer implements Summarizer {
+
+  /* Helper function that calculates the various statistics that is used for the Collector methods.*/
+  private static class LengthStats {
+    private long min = Long.MAX_VALUE;
+    private long max = Long.MIN_VALUE;
+    private long sum = 0;
+    private long[] counts = new long[32];
+
+    private void accept(int length) {
+      int idx;
+
+      if (length < min) {
+        min = length;
+      }
+
+      if (length > max) {
+        max = length;
+      }
+
+      sum += length;
+
+      if (length == 0) {
+        idx = 0;
+      } else {
+        idx = IntMath.log2(length, RoundingMode.HALF_UP);
+      }
+
+      counts[idx]++;
+    }
+
+    void summarize (String prefix, StatisticConsumer sc) {
+      sc.accept(prefix+".min", (min != Long.MAX_VALUE ? min:0));
+      sc.accept(prefix+".max", (max != Long.MIN_VALUE ? max:0));
+      sc.accept(prefix+".sum", sum);
+
+      for (int i = 0; i < counts.length; i++) {
+        if (counts[i] > 0) {
+          sc.accept(prefix+".logHist."+i, counts[i]);
+        }
+      }
+    }
+
+  }
+
+  /* Helper function for merging that is used by the Combiner. */
+  private static void merge(String prefix, Map<String, Long> stats1, Map<String,Long> stats2 ) {
+    if (stats2.containsKey(prefix+".min") && (stats1.containsKey(prefix+".min") == false)) {
+      stats1.put(prefix+".min", stats2.get(prefix+".min"));
+    } else {
+      stats1.merge(prefix+".min", stats2.get(prefix+".min"), Long::min);
+    }
+
+    if (stats2.containsKey(prefix+".max") && (stats1.containsKey(prefix+".max") == false)) {
+      stats1.put(prefix+".max", stats2.get(prefix+".max"));
+    } else {
+      stats1.merge(prefix+".max", stats2.get(prefix+".max"), Long::max);
+    }
+
+    if (stats2.containsKey(prefix+".sum") && (stats1.containsKey(prefix+".sum") == false)) {
+      stats1.put(prefix+".sum", stats2.get(prefix+".sum"));
+    } else {
+      stats1.merge(prefix+".sum", stats2.get(prefix+".sum"), Long::sum);
+    }
+
+    /* i must be less than 32 since counts[] size max is 32. */
+    for (int i = 0; i < 32; i++) {
```

Comment:
Not quite sure if 32 is correct here, if someone can confirm would be appreciated.